lib/src/parser.c

```diff
@@ -424,14 +424,15 @@ static Subtree ts_parser__lex(
       );
       ts_lexer_start(&self->lexer);
       ts_parser__restore_external_scanner(self, external_token);
+      self->lexer.data.force_serialize_state = false;
       bool found_token = self->language->external_scanner.scan(
         self->external_scanner_payload,
         &self->lexer.data,
         valid_external_tokens
       );
       ts_lexer_finish(&self->lexer, &lookahead_end_byte);
 
-      if (found_token) {
+      if (found_token || self->lexer.data.force_serialize_state) {
```

Comment:
Note: `ts_parser__lex` is a complex beast, and I'm far from certain that this is perfect. Tests seem happy with this - but I'm more than willing to take any suggestions here.